### DISCUSSION TITLE
File naming option

### DISCUSSION THREAD

**Discussion by jimpfaff** [gavvvr/obsidian-imgur-plugin#52] [Getting help]
I appreciate the plugin. It has been very helpful. I do have a request, though. Maybe I have missed the option. But I would like to be able to save images often with a particular name  when I save them remotely to Imgur or locally. I need this for future context if I reuse the image later. It's just a bit cumbersome to go into the files folder I've set up in Obsidian and wade through to figure out what is what when there is a generic name.

**Comment by gavvvr** [comment]
Hi @jimpfaff 

I am struggling to understand your use-case...
Could you please elaborate more?

Remote images have the following Markdown code:
```
![](https://i.imgur.com/m3RpPCVm.png)
```

Where would you like to have a name?

```
![here?](https://i.imgur.com/m3RpPCVm.png)
```

**Comment by jimpfaff** [comment]
I'm actually pointing more to the local save.  Note the generic names that are created on a local save in these pics. Then I renamed a couple for my purposes. It makes it easier to find them later if I want to use them in another note.

![image](https://github.com/gavvvr/obsidian-imgur-plugin/assets/23179321/cef9eb6a-1f82-487d-850b-24b1333e0b64)

![image](https://github.com/gavvvr/obsidian-imgur-plugin/assets/23179321/7c36c256-f4ca-433b-aaf2-0b295bc95856)

**Comment by gavvvr** [comment]
Hi @jimpfaff 

Local paste logic has nothing to do with this plugin. It's a built-in behavior of Obsidian.
If you want to give a file name on local paste, it's should be a job of some other plugin :)
I guess you just brought an idea for a new plugin or for the core functionality of Obsidian

**Comment by jimpfaff** [comment]
I came to you first because I figured maybe you had already seen that path but had just defaulted to the standard Obsidian functionality. I guess I'm off to Obsidian. md with my feature request! :)

**Comment by jimpfaff** [comment]
thanks for your help
